editor-lsp/src/main/java/io/github/rosemoe/sora/lsp/events/document/ApplyEditsEvent.kt

```diff
@@ -37,32 +37,33 @@ class ApplyEditsEvent : EventListener {
     override val eventName: String = "textDocument/applyEdits"
 
     override fun handle(context: EventContext) {
-        val editList: List<TextEdit> = context.get("edits")
+        val editList: List<ArrayList<TextEdit>> = context.get("edits")
         val content = context.getByClass<Content>() ?: return
 
-        editList.forEach { textEdit: TextEdit ->
-            val range = textEdit.range
-            val text = textEdit.newText
-            var startIndex =
-                content.getCharIndex(range.start.line, range.start.character)
-            var endIndex =
-                content.getCharIndex(range.end.line, range.end.character)
-            if (endIndex < startIndex) {
-                Logger.instance(this.javaClass.name)
-                    .w(
-                        "Invalid location information found applying edits from %s to %s",
-                        range.start,
-                        range.end
-                    )
-                val diff = startIndex - endIndex
-                endIndex = startIndex
-                startIndex = endIndex - diff
+        editList.forEach { list: ArrayList<TextEdit> ->
```

Comment:
No changes needed here. Consider modifying additionalTextEdits at https://github.com/Rosemoe/sora-editor/blob/ce8de8e1af88abacf807068ecedf7b9b2521aa15/editor-lsp/src/main/java/io/github/rosemoe/sora/lsp/editor/completion/LspCompletionItem.kt#L136 to remove `listOf`, and then test to see if it works properly.